jerry-core/api/jerry.c

```diff
@@ -5110,6 +5110,100 @@ jerry_get_container_type (const jerry_value_t value) /**< the container object *
   return JERRY_CONTAINER_TYPE_INVALID;
 } /* jerry_get_container_type */
 
+/**
+ * Perform container operation on the given operands (add, get, set, has, delete, size, clear).
+ *
+ * @return error - if argument has an error flag or operation is unsuccessful or unsupported
+ *         result of the container operation - otherwise
+ */
+jerry_value_t
+jerry_container_operation (jerry_container_operation_t operation, /**< container operation */
+                           jerry_value_t container, /**< container */
+                           jerry_value_t *arguments, /**< list of arguments */
+                           uint32_t arguments_number) /**< number of arguments */
+{
+  jerry_assert_api_available ();
+
+  if ((arguments_number >= 1 && ecma_is_value_error_reference (arguments[0]))
+      || (arguments_number >= 2 && ecma_is_value_error_reference (arguments[1])))
+  {
+    return jerry_throw (ecma_raise_type_error (ECMA_ERR_MSG (error_value_msg_p)));
+  }
+
+  if (ecma_is_value_object (container))
```

Comment:
So if the input `container` is no an object the error message would be that the containers are disabled?

Reply:
We could use an early return here.